### ISSUE TITLE
Add GPLv2+ and GPLv3+ as license options

### ISSUE THREAD

**Issue by connorshea** [benbalter/licensee#97]
[Dolphin](https://dolphin-emu.org/blog/2015/05/25/relicensing-dolphin/), [PPSSPP](https://github.com/hrydgard/ppsspp#ppsspp---a-fast-and-portable-psp-emulator), and [MAME](https://github.com/mamedev/mame/blob/master/LICENSE.md), for example, are all using "GPL version 2 or later". These are technically distinct choices from "just" GPLv2, as they allow derivatives to use any future version of the GPL.

This was a problem brought up by the GNU Project, so I thought it was an important consideration to suggest.

From [GPLv2](http://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html):

"9. The Free Software Foundation may publish revised and/or new versions of the General Public License from time to time. Such new versions will be similar in spirit to the present version, but may differ in detail to address new problems or concerns.

Each version is given a distinguishing version number. If the Program specifies a version number of this License which applies to it and "any later version", you have the option of following the terms and conditions either of that version or of any later version published by the Free Software Foundation. If the Program does not specify a version number of this License, you may choose any version ever published by the Free Software Foundation."

From [GPLv3](http://www.gnu.org/licenses/gpl.html):

"14. Revised Versions of this License.

The Free Software Foundation may publish revised and/or new versions of the GNU General Public License from time to time. Such new versions will be similar in spirit to the present version, but may differ in detail to address new problems or concerns.

Each version is given a distinguishing version number. If the Program specifies that a certain numbered version of the GNU General Public License “or any later version” applies to it, you have the option of following the terms and conditions either of that numbered version or of any later version published by the Free Software Foundation. If the Program does not specify a version number of the GNU General Public License, you may choose any version ever published by the Free Software Foundation.

If the Program specifies that a proxy can decide which future versions of the GNU General Public License can be used, that proxy's public statement of acceptance of a version permanently authorizes you to choose that version for the Program.

Later license versions may give you additional or different permissions. However, no additional obligations are imposed on any author or copyright holder as a result of your choosing to follow a later version."

**Comment by connorshea** [comment]
Almost forgot: AGPL 3-or-later as well. My understanding is that we can literally just copy-paste the existing licenses and append "or later" to the header and they'll be good to go.

**Comment by mlinksva** [comment]
> My understanding is that we can literally just copy-paste the existing licenses and append "or later" to the header and they'll be good to go.

That's an interesting idea. Do I understand correctly you mean like:

> GNU GENERAL PUBLIC LICENSE
> Version 3 **or later**, 29 June 2007

Neither Google nor GitHub searches turn up any instances of "Version 3 or later, 29 June 2007".

If that'd work it'd probably be a nice solution. I'm aware of the problem, which has been raised a few times eg https://github.com/github/choosealicense.com/issues/5 and https://www.reddit.com/r/programming/comments/1icwbb/choosealicensecom_from_the_good_folks_at_github/cb3ctel

I'd thought of using [other annotations](https://github.com/benbalter/licensee/issues/85#issuecomment-192719864) to indicate + in a way that choosealicense.com could hand out and licensee recognize. I suspect anything that doesn't require parsing notices where "or later" are usually indicated (which is beyond the capability of these two projects to hand out and parse at this time) would merit feedback from licensing@fsf.org which is what I intended to ask for if I had a plausible solution. Maybe this is it, but I'm wary, as the *GPLvx.y license texts aren't intended to be modified.

**Comment by connorshea** [comment]
@mlinksva Lots of results here: https://github.com/search?utf8=%E2%9C%93&q=License+Version+2+or+later&type=Code&ref=searchresults

**Comment by connorshea** [comment]
@mlinksva on a second read, I'm actually wrong. The license itself shouldn't say that, only the license name, as the "or later" is already included in the body of the license.

From [this email](http://lists.gnu.org/archive/html/repo-criteria-discuss/2016-04/msg00069.html) from Mike Gerwitz of the FSF:

> > So to clarify/break down the requirements:
> >
> > 1. "GPL v2" should be changed to "GPL v2 or later", and ditto for "GPL v3".
> > The license text should also be changed to reflect this.
> 
> In the license listing, yes, "or later" should be added.
> 
> With regards to the "license text"---do you mean the text that'll be
> placed in `LICENSE'?  If so, then no: that should be the GPL
> verbatim.  The license itself isn't saying "or later"---it's the actual
> notice on the source files themselves that says "you can use the version
> of the GPL that I have provided, or you can use a future version
> published by the FSF".
> 
> So you should recommend that "or, at your option, any later version" be
> included, as recommended here:
> 
> https://www.gnu.org/licenses/gpl.html

TL;DR The request should actually be to change the name of the license, not the license body itself. Sorry! (I blame their terrible mailing list site formatting at least partially)
